docs/guides/testing.md

```diff
@@ -1,105 +1,202 @@
+```ts
+// __mocks__/zustand.ts
+import { beforeEach, vi } from 'vitest'
+import type { StateCreator } from 'zustand'
+import { act } from '@testing-library/react'
 
-```js
-import { act } from '@testing-library/react-native'
-const { create: actualCreate } = jest.requireActual('zustand')
+const { create: actualCreate } = await jest.importActual('zustand')
 
 // a variable to hold reset functions for all stores declared in the app
-const storeResetFns = new Set()
+export const storeResetFns = new Set<() => void>()
 
 // when creating a store, we get its initial state, create a reset function and add it in the set
-export const create = (createState) => {
-  const store = actualCreate(createState)
+export const create = (<T extends unknown>(stateCreator: StateCreator<T>) => {
+  const store = actualCreate(stateCreator)
   const initialState = store.getState()
-  storeResetFns.add(() => store.setState(initialState, true))
+  storeResetFns.add(() => {
+    store.setState(initialState, true)
+  })
   return store
-}
+}) as typeof actualCreate
 
 // Reset all stores after each test run
 beforeEach(() => {
-  act(() => storeResetFns.forEach((resetFn) => resetFn()))
+  act(() => {
+    storeResetFns.forEach((resetFn) => {
+      resetFn()
+    })
+  })
 })
 ```
 
-If the `jest.config.js` has `automock: false`, then you need to do the following in `jest.setup.js`:
+> **Note:** you can remove `import { beforeEach, vi } from 'vitest'` when [globals configuration](https://vitest.dev/config/#globals)
+> is enabled
 
-```js
-jest.mock('zustand')
+```ts
+// setup-vitest.ts
+import { vi } from 'vitest'
+vi.mock('zustand') // to make it works like `Jest` (auto-mocking)
 ```
+
+> **Note**: you can remove `import { vi } from 'vitest'` when [globals configuration](https://vitest.dev/config/#globals)
+> is enabled
+
+```ts
+// vite.config.ts or vitest.config.ts
+import { defineConfig } from 'vite'
+// import { defineConfig } from 'vitest/config' // if you are using vitest without vite
+
+export default defineConfig({
+  test: {
+    // ...
+    setupFiles: ['setup-vitest.ts'],
+  },
+})
+```
+
+In the next examples we are going to use `useStore`
+
+> **Note**: all of these examples are written using `TypeScript`
+
+```ts
+// stores/user-store.ts
+import { create } from 'zustand'
+
+export type Store = {
+  count: number
+  inc: () => void
+}
+
+export const useStore = create<Store>()((set) => ({
+  count: 1,
+  inc: () => set((state) => ({ count: state.count + 1 })),
+}))
+```
+
+### Testing components
+
+```tsx
+// components/counter.tsx
+import { useStore } from '../stores/user-store'
+
+function Counter() {
+  const { count, inc } = useStore()
+
+  return (
+    <div>
+      <span>{count}</span>
+      <button onClick={inc}>one up</button>
+    </div>
+  )
+}
+```
+
+### Testing stores
+
+Still in progres. TBD
```

Comment:
Lets remove this (I know this PR is still draft, but just a reminder).

Reply:
Yeah, I'll add it